expression/builtin_op_vec.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+func (b *builtinLogicOrSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	if err := b.args[0].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	i64s := result.Int64s()
+	arg1s := buf.Int64s()
+
+	for i := 0; i < n; i++ {
+		isNull0 := result.IsNull(i)
+		isNull1 := buf.IsNull(i)
+		// Because buf is used to store the conversion of args[0] in place, it could
+		// be that args[0] is null and args[1] is nonzero, in which case the result
+		// is 1. In these cases, we need to clear the null bit mask of the corresponding
+		// row in result.
+		// See https://dev.mysql.com/doc/refman/5.7/en/logical-operators.html#operator_or
+		isNull := false
+		if (!isNull0 && i64s[i] != 0) || (!isNull1 && arg1s[i] != 0) {
+			i64s[i] = 1
+		} else if isNull0 || isNull1 {
+			isNull = true
```

Comment:
Can we delete 47, 55 lines and change this line to ```result.SetNull(i, true)```

Reply:
I believe that complicates the code a bit, because we then need to call `SetNull` in multiple branches. 

Think about the following scenario:

arg0 = NULL, arg1 = 1

In this case, the output is 1 and it goes to line 49, so we still need to call `result.SetNull(i, false)` to clear the null bitmap.